refactor-context-to-usereducer

balzhan-kanatbek opened the pull request:
- Added FontAwesome dependencies for icon usage in the application.
- Updated App component to include an "About" page.
- Refactored QuoteCard and MainPage components to improve layout and styling.
- Implemented a quotes reducer for better state management of liked quotes.
- Enhanced ProfilePage to display liked quotes with dislike functionality.
- Fixed quote formatting issues in quotes.ts for consistency.

balzhan-kanatbek committed:
feat: integrate FontAwesome icons and enhance UI components

- Added FontAwesome dependencies for icon usage in the application.
- Updated App component to include an "About" page.
- Refactored QuoteCard and MainPage components to improve layout and styling.
- Implemented a quotes reducer for better state management of liked quotes.
- Enhanced ProfilePage to display liked quotes with dislike functionality.
- Fixed quote formatting issues in quotes.ts for consistency.

balzhan-kanatbek committed:
feat: enhance quote management and UI components

- Added proxy configuration for API requests in package.json.
- Refactored App component to include Sign Up and Login pages.
- Implemented CreateQuoteForm component for adding new quotes.
- Updated QuotesContextProvider to manage liked quotes and UI state.
- Enhanced MainPage to support quote creation and improved fetching logic.
- Refactored QuoteCard to utilize liked quotes context for like/dislike functionality.
- Cleaned up quotes.ts by removing hardcoded quotes for dynamic fetching.

balzhan-kanatbek committed:
docs: update README to reflect new features and enhancements

balzhan-kanatbek committed:
docs: update README.md

annasvst commented on src/App.tsx:
```diff
@@ -5,22 +5,104 @@ import { MainPage } from "./pages/MainPage/index";
+          </button>
+        </div>
+
+        {/* Backdrop for mobile menu */}
```
Always clean up comments. If it's hard to tell what this part of code is about, you can create a new component / constant and name it:
```JS

const BackdropForMobileMenu = ({onClick}) => {
  return (
  <div className="fixed inset-0 bg-black bg-opacity-50 z-30 sm:hidden"
            onClick={() => onClick(false)}
          ></div>
  )
}
```

annasvst commented on src/QuotesContextProvider.tsx:
I think maintaining state of quotes in 3 different (or even 6 if we count both context and dispatch context) is too much. 
You can keep the quotes and information about likes in the same context. You could even add the index of the currently displayed quote to the same context, then the structure would be like this:
{quotes:[
{quote: "Some quote",
author: "Some author",
likeCount: 0,
}
],
currentQuoteIndex: 12}

balzhan-kanatbek committed:
refactor: streamline quotes context and reducer for improved state management

- Simplified QuotesContextProvider by removing unused state and context for liked quotes.
- Enhanced quotesReducer to manage additional state properties like currentQuoteIndex and showQuote.
- Updated components to utilize the new context structure, improving overall state handling.
- Refactored QuoteCard and MainPage to align with the new state management approach.

balzhan-kanatbek committed:
chore: update .gitignore to include .env file for environment variable management

balzhan-kanatbek committed:
Remove .env from repo

balzhan-kanatbek committed:
Merge branch 'main' of https://github.com/Baljann/random-quote-app into refactor-context-to-usereducer
